src/index.ts

```diff
@@ -0,0 +1,89 @@
+import Logs from './modules/logs'
+import Messages from './modules/messages'
+import Notifications from './modules/notifications'
+import Profile from './modules/profile'
+import Schemas from './modules/schemas'
+import Threads from './modules/threads'
+import Tokens from './modules/tokens'
+import Utils from './modules/utils'
+import Wallet from './modules/wallet'
+import { ApiOptions } from './models'
+
+/**
+ * Textile is the main client class
+ *
+ * @param {ApiOptions} options The API options object
+ */
+class Textile {
+  opts: ApiOptions
+  /** @property {Account} account - Manage Textile wallet account */
+  account: Account
+  /** @property {Cafes} account - Manage Textile Cafes */
+  cafes: Cafes
+  /** @property {Config} config - Manage Textile Config settings */
+  config: Config
+  /** @property {Comments} comments - Manage Textile block Comments */
+  comments: Comments
+  /** @property {Contacts} contacts - Manage Textile peer Contacts */
+  contacts: Contacts
+  /** @property {Feed} feed - Manage the Textile Feed */
+  feed: Feed
+  /** @property {Files} files - Manage Textile Files */
+  files: Files
+  /** @property {Invites} invites - Manage Textile Invites */
+  invites: Invites
+  /** @property {IPFS} ipfs - Manage the underlying IPFS peer */
+  ipfs: IPFS
+  /** @property {Likes} likes - Manage Textile block Likes */
+  likes: Likes
+  /** @property {Logs} logs - Manage Textile subsystem logs */
+  logs: Logs
+  /** @property {Messages} messages - Manage Textile thread Messages */
+  messages: Messages
+  /** @property {Notifications} notifications - Manage Textile Notifications */
+  notifications: Notifications
+  /** @property {Profile} profile - Manage a Textile node's public profile */
+  profile: Profile
+  /** @property {Schemas} schemas - Manage Textile Schemas */
+  schemas: Schemas
+  /** @property {Threads} threads - Manage Textile Threads */
+  threads: Threads
+  /** @property {Tokens} tokens - Manage Textile Threads */
+  tokens: Tokens
+  /** @property {Utils} utils - Get information about the Textile node */
+  utils: Utils
+  constructor(options: ApiOptions) {
+    this.opts = options || {}
+    this.account = new Account(this.opts)
+    this.cafes = new Cafes(this.opts)
+    this.config = new Config(this.opts)
+    this.comments = new Comments(this.opts)
+    this.contacts = new Contacts(this.opts)
+    this.feed = new Feed(this.opts)
+    this.files = new Files(this.opts)
+    this.invites = new Invites(this.opts)
+    this.ipfs = new IPFS(this.opts)
+    this.likes = new Likes(this.opts)
+    this.logs = new Logs(this.opts)
+    this.messages = new Messages(this.opts)
+    this.notifications = new Notifications(this.opts)
+    this.profile = new Profile(this.opts)
+    this.schemas = new Schemas(this.opts)
+    this.threads = new Threads(this.opts)
+    this.tokens = new Tokens(this.opts)
+    this.utils = new Utils(this.opts)
+  }
+}
+
+module.exports = { Textile, Wallet }
```

Comment:
Lemme know what you think @andrewxhill. But to 'match' more closely with the react-native-sdk API. We could export `Textile` here as API, and Wallet as Textile. Because that fits a bit more closely with the workflow in RN SDK? Just occured me to while reviewing this PR?

Reply:
yeah, this is a mess up / wrong way to export in ts anyway. surprised lint allowed that one. 

feels like we should just organize the folder structure + the api index.ts / generation the exact same as react native, making it super easy to jump between the two internally

Reply:
until it changes in RN also, you could do Textile.setup(opts) with those same otps you are passing everywhere. Again, just matching RN for now

Reply:
Seems reasonable. There are some API differences, but mostly just additional endpoints more than anything, so should work quite nicely.

Reply:
okay, fixed the export statement to be ts'y at least